[leezhanpeng] iP

leezhanpeng opened the pull request:
# Robert
> "TRACK IT DOWN! TRACK IT DOWN!!" - Lee Zhan Peng

Robert frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/leezhanpeng/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you're a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

leezhanpeng committed:
Give chatbot a new name and implement an initial skeletal version

leezhanpeng committed:
Enable Robert to echo user commands

leezhanpeng committed:
Add the ability to store and display user text

leezhanpeng committed:
Enable marking and unmarking of tasks as done using Task class

leezhanpeng committed:
Add support for ToDos, Events, and Deadline tasks via Task inheritance

leezhanpeng committed:
Add automated text UI testing

leezhanpeng committed:
Add error handling using custom RobertException

leezhanpeng committed:
Add support for deleting tasks and use ArrayList to store tasks

leezhanpeng committed:
Enable saving of tasks to hard disk

leezhanpeng committed:
Merge branch 'branch-Level-7'

leezhanpeng committed:
Make Robert understand Dates and Add clear and filter commands

leezhanpeng committed:
Merge branch 'branch-level-8'

leezhanpeng committed:
Use more OOP for Robert

leezhanpeng committed:
Merge branch 'branch-A-MoreOOP'

leezhanpeng committed:
Divide classes into packages

leezhanpeng committed:
Merge branch 'branch-A-Packages'

leezhanpeng committed:
Remove data folder from src

leezhanpeng committed:
Merge remote-tracking branch 'origin/add-gradle-support'

leezhanpeng committed:
Automate project builds using Gradle

leezhanpeng committed:
Automate project builds using Gradle

leezhanpeng committed:
Merge branch 'branch-A-Gradle'

leezhanpeng committed:
Add JUnit tests

leezhanpeng committed:
Merge branch 'branch-A-JUnit'

leezhanpeng committed:
Package the App as a JAR file

leezhanpeng committed:
Add JavaDoc comments

leezhanpeng committed:
Tweak the code to comply with checkstyle

leezhanpeng committed:
Add Find command for users to find tasks with keyword

leezhanpeng committed:
Add JavaDoc to FindCommand

leezhanpeng committed:
Merge branch 'branch-A-CodingStandard'

leezhanpeng committed:
Resolve merge conflict between A-JavaDoc and A-CodingStandard

leezhanpeng committed:
Resolve merge conflict between master and Level-9

leezhanpeng committed:
Add JavaDoc for Ui

leezhanpeng committed:
Tweak the code to further comply with checkstyle

leezhanpeng committed:
Set up checkstyle

leezhanpeng committed:
Add GUI to the chatbot

leezhanpeng committed:
Add GUI functions

saltedfishxx commented on src/main/java/robert/Robert.java:
```diff
@@ -0,0 +1,88 @@
+                + "What can I do for you today?";
+    }
+
+    public String getErrorReadingFileMessage() {
+        return "WARNING: Your previously stored tasks seems to be corrupted.\n"
+                + "As a result, your previous lists of tasks will now be cleared. Apologies!";
+    }
+
```
Good indentation for breaking of long strings, most developers will miss out the consistent indentation here!

saltedfishxx commented on src/main/java/robert/task/Deadline.java:
```diff
@@ -0,0 +1,68 @@
+        return this.byDate.equals(date);
+    }
+
+    /**
+     * Returns the string representation of the deadline task.
+     *
+     * @return the string representation.
+     */
+    @Override
+    public String toString() {
+        return String.format("[D]%s (by: %s)", super.toString(),
+                this.byDate.format(DateTimeFormatter.ofPattern("MMM dd yyyy")));
+    }
+}
```
For `@return` tags, Javadoc is actually optional for members like `toString()` as they are pretty [self-explanatory](https://google.github.io/styleguide/javaguide.html#s7.3.1-javadoc-exception-self-explanatory), hence it is okay to give a one-line summary which in this case is:

> Returns the string representation of the deadline task.

leezhanpeng replied:
Noted on this. Thanks!

saltedfishxx commented on src/main/java/robert/storage/Storage.java:
```diff
@@ -0,0 +1,136 @@
+        return downloadedTasks;
+    }
+
+    /**
+     * Saves the tasks from Robert into the hard disk.
+     *
+     * @throws RobertException if there is an issue saving the tasks into the hard disk.
+     */
+    public void save(TaskList tasks) throws RobertException {
+        try {
+            FileWriter fw = new FileWriter(this.tasksFile.toString(), false);
+            for (Task task : tasks) {
+                String storedLine;
+                String taskDone = task.getStatusIcon().equals("X") ? "1" : "0";
```
Since this method has a parameter `TaskList tasks`, you can add a `@param` tag to describe the parameter that is being passed into.

leezhanpeng replied:
Oh I missed this. Thanks a lot!

saltedfishxx reviewed (commented):
Overall good coding standard, with some minor hiccups, keep up the good work!

leezhanpeng committed:
Adjust some javadoc comments

leezhanpeng committed:
Modify build.gradle file for JAR file generation

leezhanpeng committed:
Add assertions

Using assertions at various points in the code allow for
documentation of important assumptions.

Let's use assert feature to conduct these documentation.

leezhanpeng committed:
Improve code quality

It is intuitive that production code needs to be of high quality.

Let's further improve the code's readability and naming of functions.

leezhanpeng committed:
Adjust parser's import lines

leezhanpeng committed:
Merge pull request #1 from leezhanpeng/branch-A-Assertions

Add assertions

leezhanpeng committed:
Merge branch 'master' into branch-A-CodeQuality

leezhanpeng committed:
Merge pull request #2 from leezhanpeng/branch-A-CodeQuality

Improve code quality

leezhanpeng committed:
Add duplicate detection for new task

There may be instances where users mistakenly input the same tasks multiple times.
With duplicates in the task list, it may cause confusion for the user, especially when
there are no dates associated to it (e.g. todo tasks).

Let's add a duplicate detection method during the addition of a new task.

This will prevent tasks of duplicate descriptions being in the list at the same time.

leezhanpeng committed:
Merge pull request #3 from leezhanpeng/branch-C-DetectDuplicates

Add duplicate detection for new task

leezhanpeng committed:
Add gradle.yml

leezhanpeng committed:
Modify code to conform to checkstyle

leezhanpeng committed:
Modify code to conform to checkstyle

leezhanpeng committed:
Modify gradle.yml

leezhanpeng committed:
Improve GUI

leezhanpeng committed:
Add more error handling

leezhanpeng committed:
Adjust main window code trivially

leezhanpeng committed:
Add UI screenshot

leezhanpeng committed:
Add user guide

leezhanpeng committed:
Adjust README.md

leezhanpeng committed:
Rename Ui.png

leezhanpeng committed:
Fix code quality

leezhanpeng committed:
Merge pull request #5 from leezhanpeng/branch-fixCodeQuality

Fix code quality

leezhanpeng committed:
Update README.md

leezhanpeng committed:
Update README.md

leezhanpeng committed:
Redirect data folder into src

leezhanpeng committed:
Adjust max input length for GUI

leezhanpeng committed:
Update Robert to support Java 11
